cpp/src/linear_assignment/hungarian.cu

```diff
@@ -35,28 +37,82 @@
 namespace cugraph {
 namespace detail {
 
+template <typename weight_t>
+weight_t default_precision()
+{
+  return 0;
+}
+
+template <>
+float default_precision()
+{
+  return float{1e-6};
+}
+
+template <>
+double default_precision()
+{
+  return double{1e-6};
+}
+
 template <typename index_t, typename weight_t>
 weight_t hungarian(raft::handle_t const &handle,
                    index_t num_rows,
                    index_t num_cols,
                    weight_t const *d_original_cost,
                    index_t *d_assignment,
-                   cudaStream_t stream)
+                   weight_t precision)
 {
-  //
-  //  TODO:  Can Date/Nagi implementation in raft handle rectangular matrices?
-  //
-  CUGRAPH_EXPECTS(num_rows == num_cols, "Current implementation only supports square matrices");
-
-  rmm::device_vector<index_t> col_assignments_v(num_rows);
-
-  // Create an instance of LinearAssignmentProblem using problem size, number of subproblems
-  raft::lap::LinearAssignmentProblem<index_t, weight_t> lpx(handle, num_rows, 1);
-
-  // Solve LAP(s) for given cost matrix
-  lpx.solve(d_original_cost, d_assignment, col_assignments_v.data().get());
-
-  return lpx.getPrimalObjectiveValue(0);
+  if (num_rows == num_cols) {
+    rmm::device_uvector<index_t> col_assignments_v(num_rows, handle.get_stream_view());
+
+    // Create an instance of LinearAssignmentProblem using problem size, number of subproblems
+    raft::lap::LinearAssignmentProblem<index_t, weight_t> lpx(handle, num_rows, 1, precision);
+
+    // Solve LAP(s) for given cost matrix
+    lpx.solve(d_original_cost, d_assignment, col_assignments_v.data());
+
+    return lpx.getPrimalObjectiveValue(0);
+  } else {
+    //
+    //  Create a square matrix, copy d_original_cost into it.
+    //  Fill the extra rows/columns with max(d_original_cost)
+    //
+    index_t n         = std::max(num_rows, num_cols);
+    weight_t max_cost = thrust::reduce(rmm::exec_policy(handle.get_stream_view()),
+                                       d_original_cost,
+                                       d_original_cost + (num_rows * num_cols),
+                                       weight_t{0},
+                                       thrust::maximum<weight_t>());
+
+    rmm::device_uvector<weight_t> tmp_cost(n * n, handle.get_stream_view());
```

Comment:
Why use _v for tmp_row_assignment_v and tmp_col_assignment_v but no _v for tmp_cost? IIRC, you use _v for vectors and better be consistent at least within a code block.